### ISSUE TITLE
[Bug]: 2 bugs in the "Artificial Intelligence" settings section

### ISSUE THREAD

**Issue by Jerome-Herbinet** [nextcloud/server#41344]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

1. When "OpenAI and LocalAI integration" isn't installed, we got a blank page ; I don't know if it's a bug or a feature, but I think that something should appear anyway, and also at least, for exemple if the "Translate" (provider of "Opus models by the University of Helsinki" model) or "Nextcloud Assistant" (which has a dedicated section that should appear if at list LocalAI is set) app is installed.
2. When "Nextcloud Assistant" isn't installed, the "Text processing" section shouldn't appear no matter which provider is installed 

Some screenshots can be provided if you want.

### Steps to reproduce

1. Make sure "OpenAI and LocalAI integration" isn't installed and that the "Translation" app ins installed and go to the AI settings page and see that it's blank
2. Make sure "Nextcloud Assistant" isn't installed and go to the AI settings page and notice that the text processing section is still here

### Expected behavior

Described in the bug description.

### Installation method

Community Web installer on a VPS or web space

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "nextcloud.mydomain.fr"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "27.1.3.2",
    "updatechecker": true,
    "overwrite.cli.url": "https:\/\/nextcloud.mydomain.fr",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "default_language": "fr",
    "default_locale": "fr_FR",
    "knowledgebaseenabled": false,
    "default_phone_region": "ISO3166-2",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_sendmailmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "465",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "activity_use_cached_mountpoints": true,
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "maintenance": false,
    "theme": "",
    "loglevel": 2,
    "updater.release.channel": "stable",
    "app_install_overwrite": [
        "occweb",
        "files_readmemd",
        "impersonate",
        "apporder",
        "quickaccesssorting",
        "workspace",
        "jitsi",
        "holiday_calendars"
    ],
    "mail_smtpsecure": "ssl"
}
```

### List of activated Apps

```shell
Enabled:
 - activity: 2.19.0
 - announcementcenter: 6.6.2
 - bruteforcesettings: 2.7.0
 - calendar: 4.5.3
 - circles: 27.0.1
 - cloud_federation_api: 1.10.0
 - comments: 1.17.0
 - contacts: 5.4.2
 - contactsinteraction: 1.8.0
 - dashboard: 7.7.0
 - dav: 1.27.0
 - deck: 1.11.0
 - external: 5.2.1
 - federatedfilesharing: 1.17.0
 - files: 1.22.0
 - files_external: 1.19.0
 - files_lock: 27.0.2
 - files_pdfviewer: 2.8.0
 - files_readmemd: 2.0.0
 - files_reminders: 1.0.0
 - files_rightclick: 1.6.0
 - files_sharing: 1.19.0
 - files_trashbin: 1.17.0
 - files_versions: 1.20.0
 - firstrunwizard: 2.16.0
 - flow_notifications: 1.7.0
 - groupfolders: 15.3.1
 - guests: 2.5.0
 - holiday_calendars: 0.3.0
 - impersonate: 1.14.0
 - integration_openai: 1.0.13
 - integration_openstreetmap: 1.0.6
 - integration_replicate: 1.0.8
 - integration_tmdb: 1.0.3
 - jitsi: 0.18.0
 - logreader: 2.12.0
 - lookup_server_connector: 1.15.0
 - mail: 3.4.4
 - metadata: 0.19.0
 - news: 24.0.0
 - nextcloud_announcements: 1.16.0
 - notes: 4.8.1
 - notifications: 2.15.0
 - oauth2: 1.15.1
 - password_policy: 1.17.0
 - passwords: 2023.10.30
 - phonetrack: 0.7.6
 - privacy: 1.11.0
 - provisioning_api: 1.17.0
 - quota_warning: 1.17.0
 - recommendations: 1.6.0
 - richdocuments: 8.2.2
 - richdocumentscode: 23.5.503
 - serverinfo: 1.17.0
 - settings: 1.9.0
 - sharebymail: 1.17.0
 - spreed: 17.1.2
 - support: 1.10.0
 - survey_client: 1.15.0
 - systemtags: 1.17.0
 - tables: 0.6.0
 - text: 3.8.0
 - theming: 2.2.0
 - theming_customcss: 1.14.0
 - translate: 1.2.0
 - twofactor_backupcodes: 1.16.0
 - twofactor_email: 2.7.4
 - twofactor_nextcloud_notification: 3.8.0
 - twofactor_totp: 9.0.0
 - updatenotification: 1.17.0
 - user_status: 1.7.0
 - viewer: 2.1.0
 - weather_status: 1.7.0
 - welcome: 1.0.10
 - workflowengine: 2.9.0
Disabled:
 - admin_audit
 - encryption
 - federation: 1.16.0
 - photos: 2.3.0
 - related_resources: 1.2.0
 - suspicious_login
 - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
Can be provided if necessary
```

### Additional info

Can be provided if necessary

**Comment by muchachagrande** [comment]
I've uninstalled OpenAI/LocalAI plugin and the "Artificial Intelligence" page started to show blank.
I solved it by manually deleting the ai options left by the plugin.

> occ config:app:set core ai.textprocessing_provider_preferences --value ""

**Comment by muchachagrande** [comment]
The first error was addressed here https://github.com/nextcloud/server/pull/40328#issue-1885818919

**Comment by pmarini-nc** [comment]
![image](https://github.com/nextcloud/server/assets/85954408/9dd66bbd-4b76-4f72-a55e-99ada652b2ae)
I'm on 29.0.0, this is what I get after disabling "Translate", not sure if this should be fixed already? Thanks

**Comment by joshtrichards** [comment]
Fixed in #40328

The second matter is not a a bug. The *Assistant* app is just one of several front-ends for various *Text Processing* back-end apps
